vlm/src/main/scala/geotrellis/contrib/vlm/RasterSource.scala

```diff
@@ -80,11 +80,15 @@ trait RasterSource extends CellGrid[Long] with Serializable {
   /** Raster pixel row count */
   def rows: Long = gridExtent.rows
 
+  protected def _reproject(crs: CRS, reprojectOptions: Reproject.Options, strate: OverviewStrategy): RasterSource
+
   /** Reproject to different CRS with explicit sampling reprojectOptions.
     * @see [[geotrellis.raster.reproject.Reproject]]
     * @group reproject
     */
-  def reproject(crs: CRS, reprojectOptions: Reproject.Options, strategy: OverviewStrategy): RasterSource
+  def reproject(crs: CRS, reprojectOptions: Reproject.Options, strategy: OverviewStrategy): RasterSource =
+    if (crs == this.crs) this
```

Comment:
Be aware that this can be very computationally expensive. See https://github.com/locationtech/geotrellis/issues/3022

Reply:
See also: https://github.com/locationtech/rasterframes/issues/134

Reply:
@metasim https://github.com/geotrellis/geotrellis-contrib/issues/208